vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1329,7 +1347,12 @@ func resourceVSphereVirtualMachineCreateClone(d *schema.ResourceData, meta inter
 		}
 	}
 	// Finally time to power on the virtual machine!
-	if err := virtualmachine.PowerOn(vm); err != nil {
+	pTimeoutStr := fmt.Sprintf("%ds", d.Get("poweron_timeout").(int))
+	pTimeout, err := time.ParseDuration(pTimeoutStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse poweron_timeout as a valid duration: %s", err)
+	}
```

Comment:
Might be worth adding an IntToString function to the structure helper since you've its been used a couple times.

Reply:
I decided to change the way the timeout is calculated, so this is no longer used.